codejam-virtual-keyboard/src/components/Keyboard/keyboard-data/code-layout.js

```diff
@@ -0,0 +1,10 @@
+const codeLayout = [
+  'Escape F1 F2 F3 F4 F5 F6 F7 F8 F9 F10 F11 F12 Delete',
```

Comment:
I think the better way is to define keys like this:
```javascript
  // object for "en-US" (it is useful to make these objects in different files)
  {
    KeyQ: {
      capital: 'Q',
      lovercase: 'q',
    },
    ...
  }

  // object for "en-RU" (it is useful to make these objects in different files)
  {
    KeyQ: {
      capital: 'Й',
      lovercase: 'й',
    },
    ...
  }
```

And using array of arrays with keys for generating layout.

Yes, your method is more compact, but method described below is more clear and useful.